pyispyb/app/extensions/auth/__init__.py

```diff
@@ -32,17 +31,29 @@
 class AuthProvider:
     """Allows to authentificate users."""
 
+    def __init__(self):
+        self.site_authentications = {}
+
     def init_app(self, app):
         """Init extension."""
 
-        module_name = settings.auth_module
-        class_name = settings.auth_class
-        config = {}
-        cls = getattr(importlib.import_module(module_name), class_name)
-        self._instance = cls()
-        self._instance.configure(config)
-
-    def get_auth(self, username, password, token):
+        auth_list = settings.auth
```

Comment:
I'm not sure what i think about multiple authorisation mechanism being enabled simultaneously. Is there a real use case for it? If you screw this up and you accidentally leave dummy enabled you expose the entire database

Reply:
Yes, we need it. In ISPyB, some users need to log-in with a proposal account that comes from LDAP. But we also need individual users to be able to log in with Keycloak.
I agree dummy can be dangerous. Maybe we can find some way to make it impossible to enable it in production. Or ask for confirmation at startup when it is enabled...

Reply:
Yep thats ok if theres a use case, we can find a way to make it safer.